### ISSUE TITLE
Indicator/disabled isses

### ISSUE THREAD

**Issue by 1cg** [bigskysoftware/htmx#2784]
@webdevcody noticed two issues w/ htmx:

* When an HX-Redirect occurs hx-indicators are removed even though the page is doing a top level redirect.  This looks funny.
* We do not properly clean out disabled elements when we snapshot history elements.

and has given us a PR here:

https://github.com/bigskysoftware/htmx/pull/2779

I think the issues are correct, but the fix in the PR needs to be handled differently:

* For the HX-Redirect, I think we should communicate that the request indicators should not be removed somehow via the responseHandler function, rather than checking the headers there (and should likely flag that on all early returns, including refresh, etc.)
* For the disabled elements, we should before we snapshot to history look up all disabled elements and if they have an internal request count, clear the disabled attribute.

I will try to implement these two fixes for the 2.0.2 release next week.

**Comment by webdevcody** [comment]
god bless you, and htmx

**Comment by 1cg** [comment]
@webdevcody do you have time to try the branch version of htmx for me?

**Comment by 1cg** [comment]
Here is the "fixed" source: https://raw.githubusercontent.com/bigskysoftware/htmx/1cg/wdc_fixes/src/htmx.js

**Comment by nexovec** [comment]
Is it even the intended behavior to keep the indicator on just because you got a redirect? It would be convenient though.

**Comment by phillip-haydon** [comment]
> Is it even the intended behavior to keep the indicator on just because you got a redirect? It would be convenient though.

Absolutely. Why would you remove it. The way it currently works is frustrating because you have to either run a patched version of htmx or show some intermediate state before redirecting which defeats the purpose of having a HX-Redirect header.

**Comment by nexovec** [comment]
> Absolutely. Why would you remove it.

For example you would remove it if you wanted to disable the button after the initial request is done(maybe when you have a separate indicator loader for redirects). And for some niche use-cases this is theoretically a breaking change, not sure if that's an issue for anyone.

This problem can currently be bypassed by re-enabling the indicator using [HX-Trigger Response Headers](https://htmx.org/headers/hx-trigger/) and a javascript listener, though I'm not sure why you would do that.

**Comment by dwasyl** [comment]
@1cg just tried out 2.0.2 with this change to the request indicators while waiting for the redirect and it works much more smoothly. A couple of users commented on it being "fixed" with this change. The old behaviour caused some confusion.

Thanks for including this change.

**Comment by kristofk01** [comment]
hey any update on getting this included in the next release?

**Comment by kristofk01** [comment]
This works great with the `htmx-indicator` class but it looks like the problem persists when using the loading-states extension.

**Comment by 1cg** [comment]
fixed in 2.0.3
